network/message/authorization.go

```diff
@@ -0,0 +1,375 @@
+			channels.DKGCommittee: {flow.RoleConsensus},
+		},
+	}
+}
+
+// GetMessageAuthConfig checks the underlying type and returns the correct
+// message auth Config.
+// Expected error returns during normal operations:
+//  * ErrUnknownMsgType : if underlying type of v does  not match any of the known message types
+func GetMessageAuthConfig(v interface{}) (MsgAuthConfig, error) {
+	switch v.(type) {
+	// consensus
+	case *messages.BlockProposal:
+		return AuthorizationConfigs[BlockProposal], nil
+	case *messages.BlockVote:
+		return AuthorizationConfigs[BlockVote], nil
+
+	// protocol state sync
+	case *messages.SyncRequest:
+		return AuthorizationConfigs[SyncRequest], nil
+	case *messages.SyncResponse:
+		return AuthorizationConfigs[SyncResponse], nil
+	case *messages.RangeRequest:
+		return AuthorizationConfigs[RangeRequest], nil
+	case *messages.BatchRequest:
+		return AuthorizationConfigs[BatchRequest], nil
+	case *messages.BlockResponse:
+		return AuthorizationConfigs[BlockResponse], nil
+
+	// cluster consensus
+	case *messages.ClusterBlockProposal:
+		return AuthorizationConfigs[ClusterBlockProposal], nil
+	case *messages.ClusterBlockVote:
+		return AuthorizationConfigs[ClusterBlockVote], nil
+	case *messages.ClusterBlockResponse:
+		return AuthorizationConfigs[ClusterBlockResponse], nil
+
+	// collections, guarantees & transactions
+	case *flow.CollectionGuarantee:
+		return AuthorizationConfigs[CollectionGuarantee], nil
+	case *flow.TransactionBody:
+		return AuthorizationConfigs[TransactionBody], nil
+	case *flow.Transaction:
+		return AuthorizationConfigs[Transaction], nil
+
+	// core messages for execution & verification
+	case *flow.ExecutionReceipt:
+		return AuthorizationConfigs[ExecutionReceipt], nil
+	case *flow.ResultApproval:
+		return AuthorizationConfigs[ResultApproval], nil
+
+	// execution state synchronization
+	case *messages.ExecutionStateSyncRequest:
+		return AuthorizationConfigs[ExecutionStateSyncRequest], nil
+	case *messages.ExecutionStateDelta:
+		return AuthorizationConfigs[ExecutionStateDelta], nil
+
+	// data exchange for execution of blocks
+	case *messages.ChunkDataRequest:
+		return AuthorizationConfigs[ChunkDataRequest], nil
+	case *messages.ChunkDataResponse:
+		return AuthorizationConfigs[ChunkDataResponse], nil
+
+	// result approvals
+	case *messages.ApprovalRequest:
+		return AuthorizationConfigs[ApprovalRequest], nil
+	case *messages.ApprovalResponse:
+		return AuthorizationConfigs[ApprovalResponse], nil
+
+	// generic entity exchange engines
+	case *messages.EntityRequest:
+		return AuthorizationConfigs[EntityRequest], nil
+	case *messages.EntityResponse:
+		return AuthorizationConfigs[EntityResponse], nil
+
+	// testing
+	case *message.TestMessage:
+		return AuthorizationConfigs[TestMessage], nil
+
+	// dkg
+	case *messages.DKGMessage:
+		return AuthorizationConfigs[DKGMessage], nil
+
+	default:
+		return MsgAuthConfig{}, fmt.Errorf("%w (%T)", ErrUnknownMsgType, v)
```

Comment:
When we have a sentinel error which benefits from including some other variable / data along with the error as context, it's worth creating an error type and dedicated constructor. For example with `ErrUnknownMsgType`, I think we would always want to include the unknown message type.
```suggestion
		return MsgAuthConfig{}, NewUnknownMsgTypeError(v)
```

Here's an example from HotStuff:
https://github.com/onflow/flow-go/blob/fba427fa1441587a7cbb1e26ffca14968aeb214f/consensus/hotstuff/model/errors.go#L73-L77

Reply:
https://github.com/onflow/flow-go/pull/2648/commits/948c6103bbef49efea284b5dd38b4e090d395c82